### ISSUE TITLE
Is it possible for ALVR to emulate Oculus Go inside Unity Editor?

### ISSUE THREAD

**Issue by RyotaHanabusa** [polygraphene/ALVR#143]
Thank you very much for providing such a wonderful app!

This is rather a feature request than an issue.

I found out that ALVR can also be used in Unity Editor. 
I'm currently developing for Oculus Go, and if ALVR can emulate the input for Oculus Go inside Unity Editor, it would help us developers a lot by removing the burden of building an app every single time for testing. 

Could you please consider adding Oculus Go emulation feature in the future updates?

Thank you for taking this into consideration.

**Comment by polygraphene** [comment]
Thank you for comment.
It's an interesting idea.

What exactly do you mean by "emulate the input"?

I can implement dummy HMD which produces dummy (constant) head rotation.
If you have it, you can send any head rotation/controller event with FreePIE.
Does it satisfy your request?

**Comment by RyotaHanabusa** [comment]
Thanks for the reply.
I'm a beginner programmer so forgive me if I'm wrong.

By emulate, I mean to behave exactly like Oculus Go would.
For example,(I'm talking about OVRCameraRig in Unity)
- When OVRPlugin.productName is called, it will return "Oculus Go".
- When OVRInput.IsControllerConnected(m_controller) is called, it will return "true".
- Return controller rotation as OVRInput.GetLocalControllerRotation(OVRInput.Controller.RTouch).
etc.

So basically, it would be very nice if all the buttons and tracking information of ALVR is translated into OVRCameraRig ready state so that the Play mode in Unity will work exactly like it would in Oculus Go. Does this make sense?

**Comment by polygraphene** [comment]
Okay. I got it.

You are currently using SteamVR on Unity Editor to debug app.
Is it correct?

To use OVRCameraRig on Unity Editor, perhaps we need to emulate Oculus Rift.
This functionality is provided by Revive.
I don't know if Revive can change OVRPlugin.productName or other parameters.

**Comment by RyotaHanabusa** [comment]
Yes, that's what I'm doing.

I will try and see if Revive work with Unity.

**Comment by aharriscrowne** [comment]
I am interested in this idea too for my Quest, but with the additional question of whether it can be done on a Mac.

**Comment by carlodelmazo** [comment]
> I am interested in this idea too for my Quest, but with the additional question of whether it can be done on a Mac.

any news about quest and unity?

**Comment by gino8080** [comment]
i'm following this

**Comment by MyelinsheathXD** [comment]
Hey! I used STEAM VR  unity plugin + ALVR.
And Editor runs in my Quest 2 via WIFI. 😊👌 


I think it's possible for Oculus GO as well !
